### PR TITLE
Allow documentation to build in Python 3

### PR DESCRIPTION
**Pull request by adamjstewart**
I recently made the switch to Python 3, and was trying to debug #5727 when I discovered that our documentation doesn't build at all if you're using Python 3. Previously, `make` would crash with:
```
  File "conf.py", line 67, in <module>
    stderr=subprocess.PIPE).communicate()[1].strip().split('.')
TypeError: a bytes-like object is required, not 'str'
```
Instead of running `spack -V` from the command line and screwing around with bytes and strings, I just copied the version directly from `spack.spack_version`.

P.S. Do we want to add documentation tests for Python 3? I am leaning towards no, as it isn't exactly mission critical.

### CODE REVIEW

File: lib/spack/docs/conf.py [hunk 1]
```diff
@@ -192,9 +187,9 @@ def setup(sphinx):
 # built documents.
 #
 # The short X.Y version.
-version = '.'.join(spack_version[:2])
+version = str(spack.spack_version.up_to(2))
 # The full version, including alpha/beta/rc tags.
-release = '.'.join(spack_version[:2])
+release = str(spack.spack_version.up_to(2))
```
Comment (adamjstewart) [inline lib/spack/docs/conf.py]: Should we use the full X.Y.Z version for this, or should we stick with X.Y?

Comment (tgamblin) [reply lib/spack/docs/conf.py]: I think x.y.z is better

**Comment by tgamblin** [comment]
> P.S. Do we want to add documentation tests for Python 3? I am leaning towards no, as it isn't exactly mission critical.

What if we only ran the doc tests in Python 3?

**Comment by adamjstewart** [comment]
> What if we only ran the doc tests in Python 3?

That wouldn't be very beneficial as ReadTheDocs uses Python 2 (I think).

**Pull request merged**